Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -2831,17 +2835,25 @@ MediaPlayer::SupportsType MediaPlayerPrivateGStreamer::supportsType(const MediaE
 #endif
     }
 
-    if (!ensureGStreamerInitialized())
-        return result;
-
-    GST_DEBUG("Checking mime-type \"%s\"", parameters.type.raw().utf8().data());
     if (parameters.type.isEmpty())
         return result;
 
     // This player doesn't support pictures rendering.
     if (parameters.type.raw().startsWith("image"_s))
         return result;
 
+#if USE(EXTERNAL_HOLEPUNCH)
+    HashSet<String> externalHolePunchTypes;
+    MediaPlayerPrivateHolePunch::getSupportedTypes(externalHolePunchTypes);
```

Comment:
Can this be done `once` only?

Reply:
`MediaPlayerPrivateHolePunch` already has a static `HashSet`, so there would be no point adding another static variable here.
If we want to avoid the copy done by `getSupportedTypes()`, we could make `mimeTypeCache()` a member (maybe with another name) to directly use the reference.

Reply:
It's ok as it is i think. Thanks